### PR TITLE
workflows: Configure and disable Markdown linter options

### PR DESCRIPTION
**Pull request by Alex-deVis**
* **Disable** `MD024`: Allow multiple headings with the same content
* **Configure** `MD033`: Allow inline HTML img tags
* **Disable** `MD035`: Allow the use of `----`
* **Configure** `MD101`: Remove `true`, `false`, and `root` from the list of keywords
* **Configure** `MD104`: ignore URLs, quoted strings, and specific terms that contain punctuation
* Fix markdownlinter errors on master.

### CODE REVIEW

File: content/chapters/compute/lecture/slides/processes.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 ---
 
+<!-- markdownlint-disable MD051 -->
```
Comment (teodutu) [inline content/chapters/compute/lecture/slides/processes.md:7]: Ditto. Change the link below to point to `threads.md` instead.

File: content/chapters/compute/lecture/slides/scheduling.md [hunk 2]
```diff
@@ -20,6 +20,8 @@
 
 #### Thread States
 
+<!-- markdownlint-disable MD051 -->
```
Comment (teodutu) [inline content/chapters/compute/lecture/slides/scheduling.md:23]: Ditto.

File: content/chapters/software-stack/lab/quiz/common-functions.md [hunk 3]
```diff
@@ -1,52 +1,64 @@
-# printf() System Call
+# Common functions
```
Comment (teodutu) [inline content/chapters/software-stack/lab/quiz/common-functions.md:1]:
```suggestion
# Common Functions
```
In addition, each question should be in a separate file. But it's fine for now. We'll split them another time.

File: content/chapters/software-stack/lab/quiz/libc.md [hunk 4]
```diff
@@ -1,34 +1,43 @@
-# malloc
+# libc
 
-## Question Text
+## malloc
```
Comment (teodutu) [inline content/chapters/software-stack/lab/quiz/libc.md:3]:
```suggestion
## `malloc()`
```

**Review by teodutu — CHANGES_REQUESTED** [review]
Nice work! I left a few small comments.

In addition to those, there some more linter failures. Most of them are `MD104` failures for sentences like this one: `Bla bla: something`. `:` counts as a punctuation mark so what comes after it should be placed on a new line like this:
```
Bla bla:
something
```

Other errors that I noticed were triggered by certain keywords that `makrdownlint` wrongly expects to be placed between quotes, like "root", or "true". Just remove the corresponding lines from [this file](https://github.com/open-education-hub/operating-systems/blob/master/.github/workflows/rules/md101.js). While you do this, also look for other false positives, such as "false", in `md101.js` as well as in the rest of the `md*.js` files.

**Comment by Alex-deVis** [comment]
I'll take a look at `MD101` and `MD104`.

> `:` counts as a punctuation mark so what comes after it should be placed on a new line

I would rather remove `:` from punctuation marks.

**Comment by teodutu** [comment]
> > `:` counts as a punctuation mark so what comes after it should be placed on a new line
> 
> I would rather remove `:` from punctuation marks.

Yes, this would be the more sensible option.

**Comment by Alex-deVis** [comment]
`MD104` considers URLs to be composed of multiple sentences because of `.` and `:` characters.
@razvand, @teodutu  is `.github/workflows/rules/md104.js` file downloaded or hand-written? If the latter is the case, I can add a check to skip URLs.

**Comment by teodutu** [comment]
> @razvand, @teodutu is `.github/workflows/rules/md104.js` file downloaded or hand-written? If the latter is the case, I can add a check to skip URLs.

It's irrelevant whether downloaded or hand-written. Go ahead and skip URLs, then see if there are any more spurious errors.

**Review by razvand — APPROVED** [review]
Fantastic work, @Alex-deVis . Let's merge this ASAP.

@teodutu , are you OK with it, so we could merge it?

**Review by teodutu — APPROVED** [review]
I'm also OK with this. Squash it and merge it.

**Pull request merged**